test/integration/tcp_proxy_integration_test.cc

```diff
@@ -682,6 +682,8 @@ TEST_P(TcpProxyIntegrationTest, TestIdletimeoutWithNoData) {
   tcp_client->waitForDisconnect();
 }
 
+// TODO(kuochunghsu) Debug apple CI failure.
+#if !defined(__APPLE__)
```

Comment:
What was the failure? I'd like to have some understanding of it before disabling a test on a platform.

Reply:
I don't have apple build on hand so here's what I can tell.

The log simply shows timeout for apple platform but that might be testing environment issue.
While I wrote the e2e test without preconnections, 
I grab a connection from pool, close the connection to return it to pool, the idle timer starts ticking but never fire alarms. 
I was hinted to use preconnections, and those timers fire alarms as expected.

Not a thread issue since everything seems on the same thread
```
[ RUN      ] IpVersions/TcpProxyIntegrationTest.TestPerClientIdletimeout/IPv4
[2022-12-15 17:33:08.949][32224][error][pool] [source/common/tcp/conn_pool.cc:116] [C4] setIdleTimer
[2022-12-15 17:33:08.949][32224][error][pool] [source/common/tcp/conn_pool.cc:116] [C5] setIdleTimer
[2022-12-15 17:33:08.949][32224][error][pool] [source/common/tcp/conn_pool.cc:116] [C7] setIdleTimer
[2022-12-15 17:33:08.949][32224][error][pool] [source/common/tcp/conn_pool.cc:109] [C7] disableIdleTimer
[2022-12-15 17:33:08.950][32224][error][pool] [source/common/tcp/conn_pool.cc:109] [C7] disableIdleTimer
[2022-12-15 17:33:08.950][32224][error][pool] [source/common/tcp/conn_pool.cc:116] [C7] setIdleTimer
[2022-12-15 17:33:09.050][32224][error][pool] [source/common/tcp/conn_pool.cc:102] [C4] per client idle timeout
[2022-12-15 17:33:09.050][32224][error][pool] [source/common/tcp/conn_pool.cc:109] [C4] disableIdleTimer
[2022-12-15 17:33:09.050][32224][error][pool] [source/common/tcp/conn_pool.cc:102] [C5] per client idle timeout
[2022-12-15 17:33:09.050][32224][error][pool] [source/common/tcp/conn_pool.cc:109] [C5] disableIdleTimer
```

Hence I have a hunch there's a profound testing glitch in specific platform which is beyond my knowledge.
If this is a concern, I am happy to make idle time out happen without specific platform.

Reply:
Are you able to determine for sure whether it's an issue with the test, or an issue with the feature?

I tried running 1000 iterations of the test on my mac and saw no failures. If we can't reproduce it locally, this could be a challenge to figure out.

Reply:
It's a testing issue since I could reproduce locally (finger crossed) and `TCPActiveClient` is never created unless I put `
  test_server_->waitForCounterGe("cluster.cluster_0.upstream_cx_total", 1);` to wait. PR is coming

Reply:
I `git push` the update. Let's see if ci is happy

Reply:
Nice. Both fails last time.
https://dev.azure.com/cncf/envoy/_build/results?buildId=123667&view=logs&j=fa3d3e18-6969-5713-c3e7-3581195704fd&t=ba370bef-ca49-52f5-0c93-7c0c0f27c465&l=38777

```
[ RUN      ] IpVersions/TcpProxyIntegrationTest.TestPerClientIdletimeout/IPv4
[external/com_google_absl/absl/flags/internal/flag.cc : 115] RAW: Restore saved value of envoy_reloadable_features_no_extension_lookup_by_name to: true
[external/com_google_absl/absl/flags/internal/flag.cc : 115] RAW: Restore saved value of envoy_reloadable_features_runtime_initialized to: false
[       OK ] IpVersions/TcpProxyIntegrationTest.TestPerClientIdletimeout/IPv4 (398 ms)
...
[ RUN      ] IpVersions/TcpProxyIntegrationTest.TestPerClientIdletimeout/IPv6
[external/com_google_absl/absl/flags/internal/flag.cc : 115] RAW: Restore saved value of envoy_reloadable_features_runtime_initialized to: false
[external/com_google_absl/absl/flags/internal/flag.cc : 115] RAW: Restore saved value of envoy_reloadable_features_no_extension_lookup_by_name to: true
[       OK ] IpVersions/TcpProxyIntegrationTest.TestPerClientIdletimeout/IPv6 (383 ms)

```